File: python/src/euler007.py
def solve():
    """
    10001st prime
    By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13,
    we can see that the 6th prime is 13.
    What is the 10 001st prime number?
    """
    def is_prime(n):
        if n < 2:
            return False
        for i in range(2, int(n**0.5) + 1):
            if n % i == 0:
                return False
        return True

    count = 0
    num = 1
    while count < 10001:
        num += 1
        if is_prime(num):
            count += 1
    return num

File: go/euler007.go
// https://projecteuler.net/problem=7

/*

By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.

What is the 10 001st prime number?

Answer: 104743

*/

package main

import "fmt"

func sieve(limit int) []bool {

    isPrime := make([]bool, limit+1)

    for i := range isPrime {

        isPrime[i] = true

    }

    isPrime[0], isPrime[1] = false, false

    for i := 2; i*i <= limit; i++ {

        if isPrime[i] {

            for j := i * i; j <= limit; j += i {

                isPrime[j] = false

            }

        }

    }

    return isPrime

}

func main() {

    limit := 200000

    primes := sieve(limit)

    count := 0

    for i := 2; i <= limit; i++ {

        if primes[i] {

            count++

            if count == 10001 {

                fmt.Println(i)

                return

            }

        }

    }

}